lib/access_check_filter.js

```diff
@@ -4,6 +4,35 @@ const mwUtil = require('./mwUtil');
 const P = require('bluebird');
 const URI = require('hyperswitch').URI;
 
+function redirect(content, location, options) {
+    const vary = content.headers.vary ? `${content.headers.vary}, origin` : 'origin';
+    return {
+        status: 302,
+        headers: Object.assign(content.headers, {
+            location,
+            'cache-control': options.redirect_cache_control || 'no-cache',
+            vary
+        }),
+        body: content.body
+    };
+}
+
+function response(hyper, req, path, titleParamName, location) {
+    const pathBeforeTitle = path.substring(0, path.indexOf(`{${titleParamName}}`));
+    const contentLocation = new URI(pathBeforeTitle, req.params, true).toString() + location;
+    return hyper.request({
+        method: req.method,
+        uri: new URI(contentLocation),
```

Comment:
@gwicke Oh, sorry, I should've written a comment here. Basically the plan is to do exactly that, but right now for providing the correct `content-location` header we rely on the content-location that will be set on the response to this request. The content-location code is not deployed yet, so we can't rely on it. 

More importantly, we would need to wait until all entries in Varnish expire before switching to it here to be able to provide correct content-location. After we merge this I will create a task for that.